src/main/java/com/health/healther/service/CouponService.java

```diff
@@ -111,5 +114,31 @@ public void cancelUseCoupon(Long couponId) {
 		coupon.useCoupon(false);
 	}
 
+	@Transactional
+	public void downloadCoupon(Long spaceId) {
+		Member member = memberService.findUserFromToken();
+
+		LocalDate expiredDt = LocalDate.now().minusDays(1);
+		LocalDate openDt = LocalDate.now().plusDays(1);
+
+		List<Coupon> couponList = couponRepository
+			.findBySpace_IdAndMember_IdAndExpiredDateIsAfterAndOpenDateIsBefore(
+				spaceId, member.getId(), expiredDt, openDt
+			);
+
+		if (!couponList.isEmpty()) {
+			throw new AlreadyDownloadCouponException("이미 발급된 쿠폰입니다.");
+		} else {
+			Optional<Coupon> optionalCoupon = couponRepository
+				.findTopBySpace_IdAndMember_IdAndExpiredDateIsAfterAndOpenDateIsBeforeAndIsUsed(
+					spaceId, null, expiredDt, openDt, false
+				);
+			optionalCoupon.orElseThrow(
+				() -> new AlreadySoldOutCouponException("쿠폰이 모두 소진 되었습니다.")
+			);
+			optionalCoupon.get().downloadCoupon(member);
+		}
```

Comment:
```suggestion
		if (!couponList.isEmpty()) {
			throw new AlreadyDownloadCouponException("이미 발급된 쿠폰입니다.");
		}
		Coupon coupon = couponRepository
			.findTopBySpace_IdAndMember_IdAndExpiredDateIsAfterAndOpenDateIsBeforeAndIsUsed(
				spaceId, null, expiredDt, openDt, false
			).orElseThrow(() -> new AlreadySoldOutCouponException("쿠폰이 모두 소진 되었습니다."));
		coupon.downloadCoupon(member);
		}
```

- if문으로 잡아서 throw를 던지니 else문은 따로 사용하지 않아도 될 것 같습니다.
- orElseThrow를 한줄로 사용하여 바로 Coupon을 return 받는게 깔끔해보입니당

Reply:
아하 그러네요 굳이 else를 쓰지 않아도 잡히겠네요!
coupon으로 orElseThrow를 사용해 null일 경우 바로 거르는 방법이 정돈되어 보입니다. 
피드백 반영하겠습니다 👍